e2e/tests/update_params_test.go

```diff
@@ -0,0 +1,285 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+
+	cometcli "github.com/cometbft/cometbft/libs/cli"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/gogoproto/proto"
+	"github.com/regen-network/gocuke"
+	"github.com/stretchr/testify/require"
+
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	tokenomicstypes "github.com/pokt-network/poktroll/x/tokenomics/types"
+)
+
+var updateParamsTxJSONTemplate = template.Must(
+	template.
+		New("txJSON").
+		Parse(`{
+	"body": {
+	  "messages": [
+		  {
+			"@type": "{{.Type}}",
+			"authority": "{{.Authority}}",
+			"params": {{.Params}}
+		  }
+	  ]
+	}
+}
+`))
+
+func (s *suite) AllModuleParamsAreSetToTheirDefaultValues(moduleName string) {
+	argsAndFlags := []string{
+		"query",
+		moduleName,
+		"params",
+		fmt.Sprintf("--%s=json", cometcli.OutputFlag),
+	}
+	res, err := s.pocketd.RunCommandOnHost("", argsAndFlags...)
+	require.NoError(s, err)
+
+	s.Log(res.Stdout)
+
+	switch moduleName {
+	case tokenomicstypes.ModuleName:
+		var tokenomicsParamsRes tokenomicstypes.QueryParamsResponse
+		s.cdc.MustUnmarshalJSON([]byte(res.Stdout), &tokenomicsParamsRes)
+
+	case prooftypes.ModuleName:
+		var proofParamsRes prooftypes.QueryParamsResponse
+		s.cdc.MustUnmarshalJSON([]byte(res.Stdout), &proofParamsRes)
+
+	default:
+		s.Fatalf("unexpected module name: (%v)", moduleName)
+	}
+}
+
+func (s *suite) AnAuthzGrantFromTheAccountToTheAccountForTheMessage(granterName, granterAddrType, granteeName, granteeAddrType string, msgType string) {
+	var granterAddr, granteeAddr string
+	nameToAddrMap := map[string]*string{
+		granterName: &granterAddr,
+		granteeName: &granteeAddr,
+	}
+	nameToAddrTypeMap := map[string]string{
+		granterName: granterAddrType,
+		granteeName: granteeAddrType,
+	}
+
+	for name, addr := range nameToAddrMap {
+		switch nameToAddrTypeMap[name] {
+		case "module":
+			*addr = authtypes.NewModuleAddress(name).String()
+		case "user":
+			*addr = s.getKeyAddress(name)
+		default:
+			s.Fatal("unexpected address type")
+		}
+	}
+
+	// Query the authz module for grants with the expected granter & grantee (authz.QueryGrantsRequest).
+	argsAndFlags := []string{
+		"query", "authz", "grants",
+		granterAddr, granteeAddr, msgType,
+		fmt.Sprintf("--%s=json", cometcli.OutputFlag),
+	}
+	res, err := s.pocketd.RunCommandOnHost("", argsAndFlags...)
+	require.NoError(s, err)
+
+	// Using s.cdc.UnmarshalJSON() with *authz.QueryGrantsResponse complains about
```

Comment:
Couldn't think of a beter solution to this (tried a few things) so let's leave as is. 

Optional nit: move out into a separate struct outside of this function.